app/views/layouts/application.html.erb

```diff
@@ -195,6 +195,22 @@
 </div>
 <script type="text/javascript" nonce="<%= content_security_policy_script_nonce %>">
 
+    // error handling for auth errors when getting user access tokens
+    if (window.SCP.userAccessToken === '' && <%= user_signed_in? %>) {
+        var errorTitle = 'Re-sign in again';
+        var errorText = '<p class="text-danger">There was a problem retrieving some credentials associated with your account - ' +
```

Comment:
Would it be simpler to just say "Your session has expired"?  'problem retrieving some credentials' sounds like a network error.

Reply:
While I take the point, that's actually incorrect in this case - the portal & client still thinks they have a valid user session, but we cannot retrieve their access token because it was either deleted or expired and we cannot regenerate a new one due to issues with the refresh token.  This was why I used the phrase "there was a problem retrieving credentials" because that's what is actually happening.  How about this:

```
There appears to be an issue with you session - please sign out and back in again to continue using Single Cell Portal.
```